src/exams/exams.service.ts

```diff
@@ -11,13 +13,29 @@ export class ExamsService {
   constructor(
     private readonly examsRepository: ExamsRepository,
     private readonly classroomsRepository: ClassroomsRepository,
+    private readonly enrollmentsRepository: EnrollmentsRepository,
+    private readonly mailService: MailService,
   ) {}
 
   async createOne(classroomId: number, createExamDto: CreateExamDto): Promise<Exam> {
     const classroom = await this.classroomsRepository.findOneOrFail({ id: classroomId });
 
     const exam = await this.examsRepository.createOne(createExamDto, classroom);
 
+    const enrollments = await this.enrollmentsRepository.find(
+      { classroom: classroom },
+      { populate: ["student", "student.user"] },
+    );
+    const studentEmails = enrollments.map((enrollment) => enrollment.student.user.email);
+
+    for (const email of studentEmails) {
+      await this.mailService.sendEmail(
+        email,
+        `New Exam Scheduled: ${createExamDto.title}`,
+        `An exam has been scheduled for the class: ${classroom.title}. Please check the details.`,
+      );
+    }
+
```

Comment:
Do we need to loop twice?
I think you can send the email when you're looping on the enrollments.